src/main/java/org/opensearch/neuralsearch/processor/chunker/DelimiterChunker.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * The implementation {@link FieldChunker} for delimiter algorithm
+ */
+public class DelimiterChunker implements FieldChunker {
+
+    public DelimiterChunker() {}
+
+    public static String DELIMITER_FIELD = "delimiter";
+
+    /**
+     * Validate the chunked passages for delimiter algorithm
+     *
+     * @param parameters a map containing parameters, containing the following parameters
+     * 1. A string as the paragraph split indicator
+     * @throws IllegalArgumentException If delimiter is not a string or empty
+     */
+    @Override
+    public void validateParameters(Map<String, Object> parameters) {
+        if (parameters.containsKey(DELIMITER_FIELD)) {
+            Object delimiter = parameters.get(DELIMITER_FIELD);
+            if (!(delimiter instanceof String)) {
+                throw new IllegalArgumentException("delimiter parameters: " + delimiter + " must be string.");
+            } else if (((String) delimiter).isEmpty()) {
+                throw new IllegalArgumentException("delimiter parameters should not be empty.");
+            }
+        }
+    }
+
+    /**
+     * Return the chunked passages for delimiter algorithm
+     *
+     * @param content input string
+     * @param parameters a map containing parameters, containing the following parameters
+     */
+    @Override
+    public List<String> chunk(String content, Map<String, Object> parameters) {
```

Comment:
We're not going to support regex chunking, right?

Reply:
Can we try to refactor this method by only accepting one `content` parameter, the parameter map can be extracted to class level fields so that to avoid retrieving values and force cast them here.

Reply:
Yes, regex chunking is not included in this PR.

Reply:
Class level fields have been updated. Please help review, thanks!